provider/pkg/provider/helm/v4/chart.go

```diff
@@ -260,9 +260,11 @@ func (r *ChartProvider) Construct(ctx *pulumi.Context, typ, name string, inputs
 		Objects:         objs,
 		ResourcePrefix:  *chartArgs.ResourcePrefix,
 		SkipAwait:       chartArgs.SkipAwait,
+		ParentOptions:   options,
```

Comment:
It seems awkward to have both `ParentOptions` and `ResourceOptions`. I think you should drop `ParentOptions` and introduce a yamlv2.GetChildOptions helper function to compute the value for line 264, the `ResourceOptions`.  In other words, why not compute the options eagerly at this point.